### PR TITLE
Rust: Add more CFG tests

### PR DESCRIPTION
**Pull request by hvitved**
Adds a bunch of test cases for functions with incorrect CFGs.

### CODE REVIEW

File: rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll [hunk 1]
```diff
@@ -64,18 +64,18 @@ predicate trivialPat(Pat p) { p instanceof WildcardPat or p instanceof IdentPat
 class AsmExprTree extends LeafTree instanceof AsmExpr { }
 
 class AwaitExprTree extends StandardPostOrderTree instanceof AwaitExpr {
-  override ControlFlowTree getChildNode(int i) { i = 0 and result = super.getExpr() }
+  override AstNode getChildNode(int i) { i = 0 and result = super.getExpr() }
```
Comment (aibaars) [inline rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll:67]: Why these changes? I don't think they're bad, just wondering why they were made in a commit with title `Rust: Add more CFG tests`.

**Review by aibaars — APPROVED** [review]

File: rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll (hunk 1 above)
Comment (hvitved) [reply rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll:67]: Without this change we do not get the dead-end inconsistency for the `while` loop test, because `WhileExpr` is not (yet) a `ControlFlowTree`.

**Comment by aibaars** [comment]
@hvitved Why did you merge this into `main`? The `Rust Language Tests Linux` job failed.

**Comment by hvitved** [comment]
> Why did you merge this into `main`? The `Rust Language Tests Linux` job failed.

D'oh, sorry. I merged on mobile; didn't see the CI failure.
